mono/metadata/marshal-ilgen.c

```diff
@@ -6297,6 +6297,16 @@ emit_managed_wrapper_ilgen (MonoMethodBuilder *mb, MonoMethodSignature *invoke_s
 		}
 	}
 
+	switch (sig->ret->type) {
+	case MONO_TYPE_GENERICINST: {
```

Comment:
I wish we could reuse `method_signature_is_blittable` from `marshal.c` for this... 

https://github.com/dotnet/runtime/blob/396886e43b183b2c4bcf0c996cbe61de7b067829/src/mono/mono/metadata/marshal.c#L3891-L3894